File: Python/BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
	# (52ms)
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        ans, queue = [], []
        if root:
            queue.append(root)
        while queue:
            now = []
            size = len(queue)
            for i in xrange(size):
                now.append(queue[i].val)
                if queue[i].left:
                    queue.append(queue[i].left)
                if queue[i].right:
                    queue.append(queue[i].right)
            queue = queue[size:]
            ans.append(now)
        return ans
        

File: JavaScript/BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (126ms)
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    var ans = [];
    var queue = [];
    if(root !== null)
        queue.push(root);
    while(queue.length > 0) {
        var now = [];
        var size = queue.length;
        while(0 < size--) {
            var node = queue.shift();
            now.push(node.val);
            if(node.left !== null)
                queue.push(node.left);
            if(node.right !== null)
                queue.push(node.right);
        }
        ans.push(now);
    }
    return ans;
};